test/integration/instances_test.go

```diff
@@ -589,6 +589,211 @@ func TestInstance_withPG(t *testing.T) {
+	}
+	defer instanceTeardown()
+
+}
+
+func createInstanceWithReservedIP(
+	t *testing.T,
+	client *linodego.Client,
+	reservedIP string,
+	modifiers ...instanceModifier,
+) (*linodego.Instance, func(), error) {
+	t.Helper()
+
+	createOpts := linodego.InstanceCreateOptions{
+		Label:    "go-test-ins-reserved-ip-" + randLabel(),
+		Region:   "us-east",
+		Type:     "g6-nanode-1",
+		Booted:   linodego.Pointer(false),
+		Image:    "linode/alpine3.17",
+		RootPass: randPassword(),
+		Interfaces: []linodego.InstanceConfigInterfaceCreateOptions{
+			{
+				Purpose:     linodego.InterfacePurposePublic,
+				Label:       "",
+				IPAMAddress: "",
+			},
+		},
+		Ipv4: []string{reservedIP},
+	}
+
+	for _, modifier := range modifiers {
+		modifier(client, &createOpts)
+	}
+
+	instance, err := client.CreateInstance(context.Background(), createOpts)
+	if err != nil {
+		return nil, func() {}, err
+	}
+
+	teardown := func() {
+		if terr := client.DeleteInstance(context.Background(), instance.ID); terr != nil {
+			t.Errorf("Error deleting test Instance: %s", terr)
+		}
+	}
+
+	return instance, teardown, nil
+}
+
+func TestInstance_CreateWithOwnedNonAssignedReservedIP(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithOwnedNonAssignedReservedIP")
+	defer teardown()
+
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err != nil {
+		t.Errorf("Unexpected error with owned non-assigned reserved IP: %v", err)
+	} else {
+		instanceTeardown()
+	}
+}
+
+func TestInstance_CreateWithAlreadyAssignedReservedIP(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithAlreadyAssignedReservedIP")
+	defer teardown()
+
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	// First, create an instance with the reserved IP
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err != nil {
+		t.Fatalf("Failed to create initial instance: %v", err)
+	}
+	defer instanceTeardown()
+
+	// Now try to create another instance with the same IP
+	_, secondInstanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err == nil {
+		t.Errorf("Expected error with already assigned reserved IP, but got none")
+		secondInstanceTeardown()
```

Comment:
Same as above move it right after `createInstanceWithReservedIP ` and before the error checks. Also, use `defer`